File: back/src/main/java/br/eng/eliseu/presente/model/ProdutoEstoque.java
package br.eng.eliseu.presente.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
@Table(name="produto_estoque") // Nome da tabela no banco
public class ProdutoEstoque {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Relacionamento com o Produto Principal
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "produto_id", nullable = false)
    private Produto produto;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "tamanho_id", nullable = false)
    private Tamanho tamanho;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "cor_id", nullable = false)
    private Cor cor;

    private BigDecimal preco;

    @Column(precision = 19, scale = 2)
    private BigDecimal quantidade;

    @Enumerated(EnumType.STRING)
    private StatusEnum status;

    private LocalDateTime criadoEm;
    private LocalDateTime alteradoEm;

    @Version
    private Long version;

    @PrePersist
    public void prePersist(){
        if(criadoEm==null) criadoEm = LocalDateTime.now();
        if(alteradoEm==null) alteradoEm = criadoEm;
    }

    @PreUpdate
    public void preUpdate(){
        alteradoEm = LocalDateTime.now();
    }
}

File: front/src/app/shared/model/produto-estoque.ts
import { StatusEnum } from '@/shared/model/enum/status.enum';
import { Cor } from '@/shared/model/cor';
import { Tamanho } from '@/shared/model/tamanho';
import { Produto } from '@/shared/model/produto';

export class ProdutoEstoque {

    id?: number;
    produto?: Produto;
    tamanho?: Tamanho;
    cor?: Cor;
    preco?: number;
    quantidade?: number;
    status?: StatusEnum;
    criadoEm?: Date;
    alteradoEm?: Date;
    version?: number;

    constructor(init?: Partial<ProdutoEstoque>) {
        if (init) {
            Object.assign(this, init);
        }
    }

}
